cf_xarray/accessor.py

```diff
@@ -168,6 +177,71 @@ def _get_groupby_time_accessor(var: Union[DataArray, Dataset], key: str) -> List
         return []
 
 
+def _get_custom_criteria(
+    obj: Union[DataArray, Dataset], key: str, criteria=None
+) -> List[str]:
+    """
+    Translate from axis or coord name to variable name
+
+    Parameters
+    ----------
+    obj : DataArray, Dataset
+        DataArray belonging to the coordinate to be checked
+    key : str, ["X", "Y", "Z", "T", "longitude", "latitude", "vertical", "time"]
+        key to check for.
+    criteria : dict
+        Criteria to use to map from variable to attributes describing the
+        variable. An example is coordinate_criteria which maps coordinates to
+        their attributes and attribute values.
+    error : bool
+        raise errors when key is not found or interpretable. Use False and provide default
+        to replicate dict.get(k, None).
+    default : Any
+        default value to return when error is False.
+
+    Returns
+    -------
+    List[str], Variable name(s) in parent xarray object that matches axis or coordinate `key`
+
+    Notes
+    -----
+    This functions checks for the following attributes in order
+       - `standard_name` (CF option)
+       - `_CoordinateAxisType` (from THREDDS)
+       - `axis` (CF option)
+       - `positive` (CF standard for non-pressure vertical coordinate)
+
+    References
+    ----------
+    MetPy's parse_cf
+    """
+
+    if isinstance(obj, DataArray):
+        obj = obj._to_temp_dataset()
+
+    if criteria is None:
+        if not OPTIONS["custom_criteria"]:
+            return []
+        criteria = OPTIONS["custom_criteria"]
+
+    if criteria is not None:
+        criteria = always_iterable(criteria, allowed=(tuple, list, set))
+
+    results: Set = set()
+    for crit in criteria:
```

Comment:
I integrated ChainMap in https://github.com/xarray-contrib/cf-xarray/pull/234/commits/0125789e332202755ea6c7ffcbd9999e6dfbb8ab

I like custom_criteria since it matches the criteria sort of format in `criteria.py`.